lib/macho/macho_file.rb

```diff
@@ -204,13 +202,8 @@ def dylib_id
     # @return [void]
     # @raise [ArgumentError] if `new_id` is not a String
     def dylib_id=(new_id)
-      if !new_id.is_a?(String)
-        raise ArgumentError.new("argument must be a String")
-      end
-
-      if !dylib?
-        return nil
-      end
+      raise ArgumentError.new("argument must be a String") unless new_id.is_a?(String)
```

Comment:
That's a style issue we can discuss and that I haven't brought up so far, but I'd prefer to split this across several lines to make the overall line length shorter (to stay within a with of 80 columns), i.e.:

```ruby
unless new_id.is_a?(String)
  raise ArgumentError.new("argument must be a String")
end
```

(And now that you're touching this code, “argument” feels a tad too generic in the error message.)